src/librustc_typeck/lib.rs

```diff
@@ -274,6 +291,23 @@ fn check_start_fn_ty(tcx: TyCtxt<'_>, start_def_id: LocalDefId) {
                         .emit();
                         error = true;
                     }
+
+                    for attr in it.attrs {
+                        if attr.check_name(sym::track_caller) {
+                            tcx.sess
+                                .struct_span_err(
+                                    attr.span,
+                                    "start is not allowed to be `#[track_caller]`",
```

Comment:
```suggestion
                                    "`start` is not allowed to be `#[track_caller]`",
```

Same below. Also, maybe this should just say "program entry point is not allowed..."? Up to you.

Reply:
Went with `start` for now